### PR TITLE
Fix lru and don't force to have both train and val splits for ConfigAPI

### PR DESCRIPTION
**Pull request by DaloroAT**
- Now we don't require to have validation and train split to use our API.
- Added tests for checking dataframe format.
- If `cache_size == 0` or `cache_size is None` we don't use `lru_cache`. It allows us to launch ConfigAPI with `spawn` multiprocessing start method.

### CODE REVIEW

File: oml/datasets/base.py [hunk 1]
```diff
@@ -112,7 +112,9 @@ def __init__(
         self.df = df
         self.transform = transform if transform else get_transforms("norm_albu")
         self.f_imread = f_imread
-        self.read_bytes_image_cached = lru_cache(maxsize=cache_size)(self._read_bytes_image)
+        self.read_bytes_image_cached = (
```
Comment (AlekseySh) [inline oml/datasets/base.py:115]: if you don't want to use cache you still have a function that has "cached" in the name 
let's rename it?

File: oml/datasets/list_dataset.py [hunk 2]
```diff
@@ -48,7 +48,9 @@ def __init__(
         self.filenames_list = filenames_list
         self.transform = transform
         self.f_imread = f_imread
-        self.read_bytes_image_cached = lru_cache(maxsize=cache_size)(self._read_bytes_image)
+        self.read_bytes_image_cached = (
```
Comment (AlekseySh) [inline oml/datasets/list_dataset.py:51]: the same ^

File: oml/datasets/triplet.py [hunk 3]
```diff
@@ -59,7 +59,9 @@ def __init__(
         self.transforms = transforms or get_normalisation_albu()
         assert isinstance(transforms, albu.Compose) or (transforms is None)
 
-        self.read_bytes_image_cached = lru_cache(maxsize=cache_size)(self._read_bytes_image)
+        self.read_bytes_image_cached = (
```
Comment (AlekseySh) [inline oml/datasets/triplet.py:62]: the same

**Review by AlekseySh — APPROVED** [review]
LGFM! Test covering in particular is really good.

I left only 1 small comment

**Comment by DaloroAT** [comment]
`*_cached` removed

**Review by AlekseySh — APPROVED** [review]
Great!

**Pull request merged**